tests/vmi_multus_test.go

```diff
@@ -787,24 +787,32 @@ var _ = Describe("[Serial]SRIOV", func() {
 		})
 
 		It("[test_id:3985]should create a virtual machine with sriov interface with custom MAC address", func() {
-			mac := "de:ad:00:00:be:ef"
-			vmi := getSriovVmi([]string{"sriov"})
+			const mac = "de:ad:00:00:be:ef"
+			const sriovNetworkName = "sriov"
+			vmi := getSriovVmi([]string{sriovNetworkName})
 			vmi.Spec.Domain.Devices.Interfaces[1].MacAddress = mac
 
 			startVmi(vmi)
 			waitVmi(vmi)
 
-			var interfaceName string
+			var interfaceName, networkName string
 			Eventually(func() error {
 				var err error
 				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 				Expect(err).NotTo(HaveOccurred())
+
+				networkName, err = getInterfaceNetworkNameByMAC(vmi, mac)
```

Comment:
Please move the new code outside the `Eventually`. The `Eventually` may mask bugs and we should keep it minimal.
Another option - combine `getInterfaceNetworkNameByMAC` and `getInterfaceNameByMAC` to one method `getInterfaceByMAC`

Reply:
about first option, we cant, its known that guest agent takes time to report data, even after the PR of Edy that will shorten the boot interval.
the cherry-pick PR fixes it, and here i add to it.

about 2nd, can you explain what it would solve ? 
i prefer if possible please 2 getters

Reply:
About the first - I believe you can. Both of the methods `getInterfaceNetworkNameByMAC` and `getInterfaceNameByMAC` will return error for exactly the same reason - if `vmi.Status.Interfaces` don't contain an interface with `mac`. So if `getInterfaceNameByMAC` returns without an error, `getInterfaceNetworkNameByMAC` for sure will return without an error as well.
Moreover, even if the methods would actually check `iface.Name`\`iface.InterfaceName` are set and return an error upon that, it would be still fine to move one of them outside the `Eventually`. The same code in vm.go sets both of them at the same time, so I'm not sure how the claim that "its known that guest agent takes time to report data" related.

About the second,
    

> - for example you can have the iface name, but not the network name (in case you know the MAC but didnt set it)
>     and then its good to have the freedom of using just one at a time.

So what? If you return the whole interface status, use whatever field you wish.

> -     Moreover and important imo, once the PR of using network data to config the ip is created, the use of getInterfaceNameByMAC might be totally removed.

I don't understand why using network data to config the ip will lead to removal of `getInterfaceNameByMAC`. But we can discuss on the PR doing it. Here we have both of the methods in use, so currently this claim is irrelevant.

Reply:
For the network name and the mac indeed they can come together, 
for the ips for examples they dont (even if nmcli right commands are used from what i remember i tried).

I will go with the first option, remove the `getInterfaceNetworkNameByMAC` from the Eventually.
(wanted it to be more explicit that its part of the values that are reported by guest agent and should be waited, once IPs for example would be added there)

Reply:
> I don't understand why using network data to config the ip will lead to removal of `getInterfaceNameByMAC`. But we can discuss on the PR doing it. Here we have both of the methods in use, so currently this claim is irrelevant.

FYI, i think that since it will use setName, so we wont need to know the interface we need to check the name of
https://github.com/kubevirt/kubevirt/pull/4448/files#diff-b03439ba2dac10103df177f90802bbfa4d14104b15a9247fa885e7d6d9bea91eR128